daemon/status.go

```diff
@@ -107,6 +109,119 @@ func (d *Daemon) getNodeStatus() *models.ClusterStatus {
 	return &clusterStatus
 }
 
+func (d *Daemon) setPrometheusMetrics() {
+	healthClient, err := healthclient.NewClient("")
+	if err != nil {
+		log.WithError(err).Error("Unable to get cilium-health client")
+		return
+	}
+	healthStatus, err := healthClient.Connectivity.GetStatus(nil)
+	if err != nil {
+		log.WithError(err).Error("Unable to get health status for Prometheus metrics")
+		return
+	}
+
+	// Controllers failing
+	controllersFailing := 0
+	if d.statusResponse.Controllers != nil {
+		for _, ctrl := range d.statusResponse.Controllers {
+			if ctrl.Status == nil {
+				continue
+			}
+			if ctrl.Status.ConsecutiveFailureCount > 0 {
+				controllersFailing++
+			}
+		}
+	}
+	metrics.ControllersFailing.Set(float64(controllersFailing))
+
+	// Address count
+	metrics.IPv4Addresses.Set(float64(len(d.statusResponse.IPAM.IPV4)))
+	metrics.IPv6Addresses.Set(float64(len(d.statusResponse.IPAM.IPV6)))
+
+	// Nodes health
+	var (
+		unreachableNodes     int64
+		unreachableEndpoints int64
+
+		sumNodeICMPLatency int64
+		maxNodeICMPLatency int64
+
+		sumNodeHTTPLatency int64
+		maxNodeHTTPLatency int64
+
+		sumEndpointICMPLatency int64
+		maxEndpointICMPLatency int64
+
+		sumEndpointHTTPLatency int64
+		maxEndpointHTTPLatency int64
+	)
+
+	nodes := healthStatus.Payload.Nodes
+
+	if healthStatus.Payload.Nodes != nil {
```

Comment:
~why not use the variable you have assigned in the previous line?~ You do not need to do this, if `healthStatus.Payload.Nodes` is `nil`, the for loop is not executed